resolwe_bio/processes/import_data/metadata.py

```diff
@@ -1,93 +1,161 @@
+    new_path = path.with_suffix(path.suffix.lower())
+    path.replace(new_path)
+    return new_path
+
+
+def read_tabular_data(path, error):
+    """Convert the uploaded file to Pandas data frame."""
+    extensions = [".csv", ".tab", ".tsv", ".xlsx", ".xls"]
+    if path.suffix not in extensions:
+        error(
+            "Unsupported file name extension. Supported extensions "
+            f"are {', '.join(extensions)}."
+        )
+    if path.suffix == ".xls":
+        return pd.read_excel(path, engine="xlrd")
+    elif path.suffix == ".xlsx":
+        return pd.read_excel(path, engine="openpyxl")
+    elif any(path.suffix == ext for ext in [".tab", ".tsv"]):
+        return pd.read_csv(path, sep="\t")
+    elif path.suffix == ".csv":
+        return pd.read_csv(path)
     else:
-        return "(no missing values)"
+        return pd.DataFrame()
 
 
-def get_features_description(data):
-    """Get formatted features description."""
-    missing_features = format_missing(
-        data.has_missing_attribute() and data.get_nan_frequency_attribute()
-    )
-    return f"{len(data.domain.attributes)} {missing_features}"
+class UploadMetadataUnique(Process):
+    """Upload metadata file where each row corresponds to a single sample.
 
+    The uploaded metadata table represents one-to-one (1:1) relation to
+    samples in the working collection. Metadata table must contain a column
+    with one of the following headers: "Sample ID", "Sample name" or "Sample slug".
+    """
 
-def get_target_description(data):
-    """Get formatted target class description."""
-    missing_in_class = format_missing(
-        data.has_missing_class() and data.get_nan_frequency_attribute()
-    )
+    slug = "upload-metadata-unique"
+    name = "Metadata table (one-to-one)"
+    process_type = "data:metadata:unique"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/s4q6j6e8/resolwebio/orange:2.0.0"}
+        },
+        "resources": {"cores": 1, "memory": 8192},
+    }
+    data_name = '{{ src.file|default("?") }}'
 
-    if data.domain.has_continuous_class:
-        target_description = f"Regression; numerical class {missing_in_class}"
-    elif data.domain.has_discrete_class:
-        target_description = (
-            "Classification; categorical class "
-            f"with {len(data.domain.class_var.values)} values {missing_in_class}"
-        )
-    elif data.domain.class_vars:
-        target_description = (
-            "Multi-target; "
-            f"{len(data.domain.class_vars)} target variables "
-            f"{missing_in_class}"
+    class Input:
+        """Input field to process UploadMetadataUnique."""
+
+        src = FileField(
+            label="Table with metadata",
+            description="The table should be in Orange format and use "
```

Comment:
No need for the table to be in Orange format. It just needs to be tabular end contain one of the required columns.